.github/workflows/build-test.yml

```
@@ -1111,9 +1111,19 @@ jobs:
           COPILOT_EXA_API_KEY: ${{ secrets.COPILOT_EXA_API_KEY }}
         uses: ./.github/actions/server-test-env
 
+      - name: Upload test results
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: test-results-e2e-server-${{ matrix.tests.shard }}
```

actionlint (1.7.7)

property "tests" is not defined in object type {shardindex: number; shardtotal: number}

(expression)